nengo/builder/connection.py

```diff
@@ -76,7 +76,11 @@ def get_targets(conn, eval_points):
     else:
         targets = np.zeros((len(eval_points), conn.size_mid))
         for i, ep in enumerate(eval_points[:, conn.pre_slice]):
-            targets[i] = conn.function(ep)
+            out = conn.function(ep)
+            if out is None:
+                raise BuildError("Building %s: Connection function returned "
```

Comment:
Personally I prefer to have the memory addresses in the error message for unlabeled objects because it is the only thing maybe allows me to identify the object. Adding a “debug mode” might make the error message to appear simpler, but it will require users to know about the debug mode and it is likely to introduce more complexity to the code base.